connector/connect/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -106,24 +113,39 @@ message Project {
   //
   // For example, `SELECT ABS(-1)` is valid plan without an input plan.
   Relation input = 1;
+
+  // (Required) A Project requires at least one expression.
   repeated Expression expressions = 3;
 }
 
 // Relation that applies a boolean expression `condition` on each row of `input` to produce
 // the output result.
 message Filter {
+  // (Required) Input relation for a Filter.
   Relation input = 1;
+
+  // (Required) A Filter must have a condition expression.
   Expression condition = 2;
 }
 
 // Relation of type [[Join]].
 //
 // `left` and `right` must be present.
 message Join {
+  // (Required) Left input relation for a Join.
   Relation left = 1;
+
+  // (Required) Right input relation for a Join.
   Relation right = 2;
+
+  // (Optional) The join condition. Could be unset when `using_columns` is utilized.
+  //
+  // This field does not co-exist with using_columns.
   Expression join_condition = 3;
```

Comment:
why is this not optional?

Reply:
This is `Expression` which is a message. By default it has `has_field_name`.

Reply:
In fact, not sure if we need to improve the proto style:

For a optional Message field, we also use `optional` to mark to keep consistency?